build/ci/compare_benchmarks.py

```diff
@@ -0,0 +1,188 @@
+            parsed = parse_bench_line(line)
+            if parsed is None:
+                raise ValueError(f"failed parse Benchmark line: '{line}'")
+
+            name_test, metrics = parsed
+            if name_test not in results:
+                results[name_test] = {m: [] for m in KNOWN_METRICS.keys()}
+
+            for metric_name, value in metrics.items():
+                results[name_test][metric_name].append(value)
+    return results
+
+
+def aggregate_results(
+        parsed_metrics: Dict[str, Dict[str, List[float]]],
+        method: Literal["mean", "median"],
+) -> Dict[str, Dict[str, float]]:
+    aggregated: Dict[str, Dict[str, float]] = {}
+    for bench_name, metrics in parsed_metrics.items():
+        aggregated[bench_name] = {}
+        for m, values in metrics.items():
+            if not values:
+                continue
+            if method == "median":
+                aggregated[bench_name][m] = statistics.median(values)
+            elif method == "mean":
+                aggregated[bench_name][m] = statistics.mean(values)
+    return aggregated
+
+
+def format_metric_changes(
+        old_val, new_val: float,
+        metric_name: str,
+        alert_threshold: float,
+        alert_emojis: Dict[str, str],
+        metric_change_interpretation: Dict[str, Dict[str, str]],
+) -> str:
+    # the metric exists only in new file
+    if old_val is None and new_val is not None:
+        return f"New metric: {new_val:.2f} ⚠️"
+
+    # the metric exists only in old file
+    if new_val is None and old_val is not None:
+        return f"Only old metric: {old_val:.2f} ⚠️"
+
+    # the metric doesn't exist for this bench
+    if old_val is None and new_val is None:
+        raise f"both old and new values are 'None' for metric {metric_name}"
+
+    change_pct = ((new_val - old_val) / old_val) * 100
+
+    better_direction = metric_change_interpretation[metric_name]['better']
+    is_better = (change_pct > 0 if better_direction == 'up' else change_pct < 0)
+
+    emoji = ""
+    if alert_threshold is not None and abs(change_pct) >= alert_threshold:
+        emoji = alert_emojis['good'] if is_better else alert_emojis['bad']
+
+    res = f"{humanize_number(old_val)} → {humanize_number(new_val)} ({change_pct:+.2f}%)"
+    return res if emoji == "" else res + f" {emoji}"
+
+
+def humanize_number(val: float) -> str:
+    abs_val = abs(val)
+    if abs_val >= 1_000_000:
+        return f"{val/1_000_000:.2f}M"
+    elif abs_val >= 1_000:
+        return f"{val/1_000:.2f}K"
+    else:
+        return f"{val:.2f}"
+
+
+def format_benchmark_name(raw_name: str) -> str:
+    name = raw_name[len("Benchmark"):] if raw_name.startswith("Benchmark") else raw_name
+    name = name.replace("/", " ")
+
+    # replace patterns key-value on key=value
+    def replace_key_value(match):
+        key, val = match.group(1), match.group(2)
+        return f"{key}={val}"
+
+    name = re.sub(r"(\w+)-(\w+)", replace_key_value, name)
```

Comment:
Maybe it's better to put params in brackets like this: `Float Random (bits=32, cpu=4)`